pytest/tests/mocknet/mirror.py

```diff
@@ -313,24 +319,33 @@ def get_network_nodes(new_test_rpc_responses, num_validators):
     boot_nodes = []
     for node, response in new_test_rpc_responses:
         if len(validators) < num_validators:
-            if node.can_validate and response[
-                    'validator_account_id'] is not None:
-                # we assume here that validator_account_id is not null, validator_public_key
-                # better not be null either
+            if node.can_validate:
+                # We assume that if node validates a chain, it has account id
+                # and public key.
+                if node.role() == 'validator':
+                    amount = 10**32
+                else:
+                    amount = 10**33
+
                 validators.append({
                     'account_id': response['validator_account_id'],
                     'public_key': response['validator_public_key'],
-                    'amount': str(10**33),
+                    'amount': str(amount),
                 })
+
+                if len(boot_nodes) < 20:
+                    boot_nodes.append(
+                        f'{response["node_key"]}@{node.ip_addr()}:{response["listen_port"]}'
+                    )
         else:
             non_validators.append(node.ip_addr())
```

Comment:
Any node that holds the initial state can be a boot node. In this case, I would avoid adding `role == validator` to the list of boot nodes. For readability, add a function `has_setup_state`.

```suggestion
        else:
            non_validators.append(node.ip_addr())
        if len(boot_nodes) < 20 and node.has_setup_state():
            boot_nodes.append(
                f'{response["node_key"]}@{node.ip_addr()}:{response["listen_port"]}'
            )
```

Reply:
Oh, I see. Well, `has_setup_state` would be trivial, so I want to just check the role and add comment.